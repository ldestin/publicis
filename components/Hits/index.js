import React from 'react'
import PropTypes from 'prop-types'
import Hit from '../Hit'
import { PRIMARY_COLOR } from '../../constants'

export const getColor = (hit, maxPoints) => {
  const ratio = hit.points > maxPoints ? 1 : hit.points / maxPoints

  return `rgb(${Math.round(PRIMARY_COLOR.RED * ratio)}, ${Math.round(PRIMARY_COLOR.GREEN * ratio)}, ${Math.round(PRIMARY_COLOR.BLUE * ratio)})`
}

const Hits = ({ hits, maxPoints, onHideClick, onUpvoteClick }) => (
  <div className="hits">
    {hits.map(hit => (
      <Hit
        key={hit.objectID}
        hit={hit}
        color={getColor(hit, maxPoints)}
        onHideClick={onHideClick}
        onUpvoteClick={onUpvoteClick}
      />
    ))}
  </div>
)

Hits.defaultProps = {
  hits: [],
  onHideClick: undefined,
  onUpvoteClick: undefined,
}

Hits.propTypes = {
  hits: PropTypes.arrayOf(PropTypes.shape({
    num_comments: PropTypes.number,
    points: PropTypes.number,
    title: PropTypes.string,
    url: PropTypes.string,
    author: PropTypes.string,
    created_at: PropTypes.string
  })),
  maxPoints: PropTypes.number.isRequired,
  onHideClick: PropTypes.func,
  onUpvoteClick: PropTypes.func,
}

export default Hits
