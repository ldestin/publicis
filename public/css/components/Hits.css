.hits {
  padding: 1rem 0;
}
