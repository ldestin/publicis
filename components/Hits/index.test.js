import React from 'react'
import { render } from '@testing-library/react'
import Hits, { getColor } from '.'

jest.mock('../Hit', () => props => <div>Hit: {JSON.stringify(props, null, 2)}</div>)

describe('getColor function', () => {
  it('should return the correct color', () => {
    const result = getColor({ points: 5 }, 10)

    expect(result).toBe('rgb(128, 51, 0)')
  })

  it('should return the correct color with max ratio', () => {
    const result = getColor({ points: 10 }, 10)

    expect(result).toBe('rgb(255, 102, 0)')
  })
})

describe('Hits component', () => {
  const props = {
    hits: [
      { objectID: '1', points: 5 },
      { objectID: '2', points: 10 }
    ],
    maxPoints: 10,
    onHideClick: jest.fn(),
    onUpvoteClick: jest.fn(),
  }

  it('shouold render correctly', () => {
    const { container } = render(<Hits {...props} />)

    expect(container).toMatchSnapshot()
  })
})
