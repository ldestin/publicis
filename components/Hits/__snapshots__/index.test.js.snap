// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits component shouold render correctly 1`] = `
<div>
  <div
    class="hits"
  >
    <div>
      Hit: 
      {
  "hit": {
    "objectID": "1",
    "points": 5
  },
  "color": "rgb(128, 51, 0)"
}
    </div>
    <div>
      Hit: 
      {
  "hit": {
    "objectID": "2",
    "points": 10
  },
  "color": "rgb(255, 102, 0)"
}
    </div>
  </div>
</div>
`;
